ambry-store/src/main/java/com/github/ambry/store/LogSegment.java

```diff
@@ -521,6 +539,7 @@ long getCapacityInBytes() {
    */
   void flush() throws IOException {
     fileChannel.force(true);
+    directIOExecutor.flush();
```

Comment:
I'm not sure if it makes any difference.
Shall we can directIOExecutor.flush() before fileChannel.force()?

Same question for the close.

Reply:
I don't actually if it's necessary to even call both of them. I suppose it's fine to do so.

Reply:
DirectIOExecutor is based on fileChannel and utilize file channel. I think it's better to flush it before fileChannel to be safe.